### PR TITLE
Decouple the metadata fetching with TaskMetadataExtract

### PR DESCRIPTION
**Pull request by deldesir**
🚀 **Pull Request Overview:**

This pull request allows videos to be downloaded in their own tasks. It calls [TaskMetadataExtract], which then calls TaskDownload for all video urls found subsequently

📋 **Checklist:**
- [x] Tested the changes thoroughly.

🔗 **Related Issue(s)**: 
Issue #104

📌 **Testing scenarios:**
See Issue #97 

cc @EMG70

PS _This PR builds upon #110_

### CODE REVIEW

**Comment by holta** [comment]
Why is this PR in DRAFT mode?

(Thanks @deldesir for explaining or resolving!)

File: cps/editbooks.py [hunk 1]
```diff
@@ -381,96 +381,89 @@ def move_mediafile(requested_files, current_user_name=None, shelf_title=None):
                 log.error("Error occurred: %s", ex)
                 flash(_("There was an error"), category="error")
 
-        log.info("Requested files list: %s", requested_files)
-        for requested_file in requested_files:
```
Comment (deldesir) [inline cps/editbooks.py:385]: Previously, we expected a list of requested files. Now files are processed in a case by case basis. Those files are sent from the TaskDownload from /cps/tasks.download.py. This is the reason here a block of code has been indented left, eliminating the previous for loop.

**Comment by holta** [comment]
FYI "Download to IIAB" of YouTube playlists/channels does not work with this PR #111 and #112.

@deldesir is trying to fix it as we speak! :sweat_smile:

**Comment by holta** [comment]
Thank you @EMG70 and @deldesir for careful TDD (focused testing) over the past week.

Looking to the future now with these PRs are others:

- PR #115
- PR #116

**Pull request merged**